fpdf/fonts.py

```diff
@@ -172,6 +258,18 @@ def __init__(
         self.font_name = None
 
 
+@dataclass(order=True, frozen=True)
+class Glyph:
+    """
+    This represents one glyph on the font
+    Unicode is a tuple because ligatures or character substitution
+    can map a sequence of unicode characters to a single glyph
+    """
+
```

Comment:
```suggestion
__slots__ = ("glyph_id", "unicode")
```
This small memory optimization seems valuable as many instances of `Glyph` will be instantiated.
Some context regarding this: https://chezsoi.org/lucas/blog/slots-memory-optimizations-in-python.html